types/protocol.d.ts

```diff
@@ -10,6 +10,8 @@ type CrdpEvents = CrdpMappings.Events;
 type CrdpCommands = CrdpMappings.Commands;
 
 declare module Protocol {
+  type TargetType = 'page' | 'iframe' | 'worker';
```

Comment:
can we switch `iframe` to `oopif` to make it explicit?

Reply:
I don't think it's a good idea to make this target type not just be the Crdp target type. Better not to make up our own terminology. If you're insistent on this, we should rename this from target type to something else.

Reply:
No, that's a good point since we're saving these in the devtoolslog messages, it makes sense to stick what's coming in over the protocol (even if we are combining a couple of things now).

I guess I'm really thinking about what we expose from NetworkRequest. Both `page` and `iframe` are misleading for what audits may be doing (the first is really page/in-process iframes, the second just out-of-process iframes). The distinction is important for audits that only care about main-frame activity (e.g. LH LCP can't be in an iframe, even if the iframe shares a process with the main frame), and being able to easily make the distinction could be very helpful to LR, where I believe all iframes are in-process, regardless of origin.

I think we may want something like `'page'|'ipif'|'oopif'|'worker'` for networkRequests? We can make that actual change in a follow up, but worth discussing the design now: would that seem workable to you even though it overlaps but doesn't quite match what would come through on the devtoolslog `targetType`?

Reply:
how about making the record's sessionTargetType a private member, and expose helpers `isFromOutOfProcessIframe`, `isFromIframe`, etc.. as audit uses cases demand?

Reply:
that could work, I was sticking to the enum because we had discussed it in #14157. It does have the nice property that these are exclusive states, so you know you don't have to check more than one boolean helper

Reply:
that could work to land this PR though. Make `sessionTargetType` private, keep `isFromOutOfProcessIframe` as is, and we can discuss building on it next week?